workload/basic_workload.cpp

```diff
@@ -140,33 +161,47 @@ BasicWorkload::NextTransaction() {
   unordered_set<Key> write_set;
   std::ostringstream code;
 
-  // Fill txn with write operators
+  auto num_writes = params_.GetUInt32(NUM_WRITES);
+  auto num_hot_records = params_.GetUInt32(NUM_HOT_RECORDS);
+  auto num_records = params_.GetUInt32(NUM_RECORDS);
   auto value_size = params_.GetUInt32(VALUE_SIZE);
-  for (size_t i = 0; i < num_writes; i++) {
+
+  CHECK_LE(num_writes, num_records)
+      << "Number of writes cannot exceed number of records in a transaction!";
+  CHECK_LE(num_hot_records, num_records)
+      << "Number of hot records cannot exceed number of records in a transaction!";
+
+  auto hot_indices = Choose(num_records, num_hot_records, re_);
```

Comment:
Could add a note that this is necessary so that the hot keys aren't always the writes